### PR TITLE
ENH: Increasing sklearn compatibility

### PR DESCRIPTION
**Pull request by replacementAI**
Adding parameter optimization compatibility

### CODE REVIEW

**Comment by replacementAI** [comment]
https://github.com/fracdiff/fracdiff/issues/148

Currently the Fracdiff class doesnt inherit from BaseEstimator class, which is inherited by [all sklearn native transformers](https://github.com/scikit-learn/scikit-learn/blob/f3f51f9b6/sklearn/preprocessing/_data.py). Doing so allows us to parameter search Fracdiff, specifically the d and window parameters which is really useful.

**Comment by replacementAI** [comment]
Waiting on approval

**Comment by simaki** [comment]
thank you for a good catch and pertinent pr @replacementAI, could you kindly recreate a pr? sorry i accidentally closed it by mistake and GitHub UI does not allow me to reopen it somehow.

File: fracdiff/sklearn/fracdiff.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
-import numpy  # type: ignore
+from typing import TypeVar
+
+import numpy
 from sklearn.base import TransformerMixin  # type: ignore
+from sklearn.base import BaseEstimator # type: ignore
```
Comment (ghost) [inline fracdiff/sklearn/fracdiff.py]:
```suggestion
from sklearn.base import BaseEstimator  # type: ignore
```

File: fracdiff/sklearn/fracdiff.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
-import numpy  # type: ignore
+from typing import TypeVar
+
+import numpy
 from sklearn.base import TransformerMixin  # type: ignore
+from sklearn.base import BaseEstimator  # type: ignore
```
Comment (ghost) [inline fracdiff/sklearn/fracdiff.py]:
```suggestion
from sklearn.base import BaseEstimator  # type: ignore
from sklearn.base import TransformerMixin  # type: ignore
```

**Comment by ghost** [comment]
thank you for a good catch and pertinent pr @replacementAI !

**Pull request merged**